attribute.go

```diff
@@ -296,3 +296,132 @@ func (ad *AttributeDecoder) Do(fn func(b []byte) error) {
 		ad.err = err
 	}
 }
+
+// An AttributeEncoder provides a safe way to encode attributes.
+//
+// It is recommended to use an AttributeEncoder where possible instead of
+// calling MarshalBinary and using package nlenc directly.
+//
+// Errors from intermediate encoding steps are returned in the call to
+// MarshalBinary.
+type AttributeEncoder struct {
```

Comment:
Do you think it makes sense to expose `func (ae *AttributeEncoder) Attrs() ([]netlink.Attribute, error)` to get the attribute stack out, in addition to or instead of returning bytes directly?

Reply:
I have no preference. I have no need for this, so didn't include it.

Reply:
I still haven't included this. Is there a specific case somewhere that needs this?